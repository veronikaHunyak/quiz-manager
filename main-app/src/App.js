import React, { Component } from "react";
import {
  BrowserRouter as Router,
  Switch,
  Route,
  Link,
  Redirect,
} from "react-router-dom";

import Login from "./components/login";

export const userTypes = {
  edit: "Edit", //with username of user1
  view: "View", //with username of user2
  restricted: "Restricted", //with username of user3
};

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      userType: localStorage.getItem("user"),
      loggedIn: false,
    };
  }

  componentDidMount() {
    if (this.state.userType) {
      this.setState({ loggedIn: true });
    } else {
      this.setState({ loggedIn: false });
    }
  }

  logOut = () => {
    this.setState({ loggedIn: false });
    localStorage.removeItem("user");
  };

  loginStateChange = (loginType) => {
    localStorage.setItem("user", loginType);
    this.setState({ loggedIn: true, userType: loginType });
  };

  renderlogin() {
    if (!this.state.loggedIn) {
      return (
        <Link to="/login" className="nav-link">
          Login
        </Link>
      );
    } else {
      return (
        <Link to="/" className="nav-link" onClick={this.logOut}>
          Logout
        </Link>
      );
    }
  }

  render() {
    return (
      <Router>
        <div>
          <nav
            className="navbar navbar-expand-lg navbar-light"
            style={{ backgroundColor: "aliceblue" }}
          >
            <Link to="/quizes" className="navbar-brand">
              Quiz Manager
            </Link>
            <button
              className="navbar-toggler"
              type="button"
              data-toggle="collapse"
              data-target="#navbarText"
              aria-controls="navbarText"
              aria-expanded="false"
              aria-label="Toggle navigation"
              style={{ backgroundColor: "whitesmoke" }}
            >
              <span className="navbar-toggler-icon"></span>
            </button>
            <div className="collapse navbar-collapse" id="navbarText">
              <ul className="navbar-nav mr-auto">
                <li className="navbar-item">
                  <Link to="/quizes" className="nav-link">
                    Quizes
                  </Link>
                </li>
                <li className="navbar-item">{this.renderlogin()}</li>
              </ul>
              {this.state.loggedIn && (
                <p>
                  Welcome back, your access rights are:{" "}
                  <b>{this.state.userType}</b>
                </p>
              )}
            </div>
          </nav>

          <Switch>
            <Route
              path="/login"
              exact
              render={(props) => (
                <Login {...props} loginStateChange={this.loginStateChange} />
              )}
            >
              {this.state.loggedIn ? (
                <Redirect to="/quizes" />
              ) : (
                <Login loginStateChange={this.loginStateChange} />
              )}
            </Route>

            {/* <Route
              path="/quizes"
              render={(props) => (
                <ViewQuizes
                  {...props}
                  loggedInUserType={this.state.userType}
                  quizes={this.state.quizes}
                />
              )}
            /> */}

            <Route path="/" exact>
              {this.state.loggedIn ? (
                <Redirect to="/quizes" />
              ) : (
                <Login loginStateChange={this.loginStateChange} />
              )}
            </Route>
          </Switch>
        </div>
      </Router>
    );
  }
}

export default App;
