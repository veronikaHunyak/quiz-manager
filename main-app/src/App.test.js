import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import App, { userTypes } from "./App";

describe("App", () => {
  it("should match snapshot if user is logged out", () => {
    const wrapper = shallow(<App />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should match snapshot if user is logged in", () => {
    const wrapper = shallow(<App />);
    wrapper.setState({ loggedIn: true });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should match snapshot if user has edit rights", () => {
    const wrapper = shallow(<App />);
    wrapper.setState({ userType: userTypes.edit, loggedIn: true });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should match snapshot if user has view rights", () => {
    const wrapper = shallow(<App />);
    wrapper.setState({ userType: userTypes.view, loggedIn: true });
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should match snapshot if user has restricted rights", () => {
    const wrapper = shallow(<App />);
    wrapper.setState({ userType: userTypes.restricted, loggedIn: true });
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
