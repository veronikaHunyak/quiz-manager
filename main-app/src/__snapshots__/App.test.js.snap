// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match snapshot if user has edit rights 1`] = `
<BrowserRouter>
  <div>
    <nav
      className="navbar navbar-expand-lg navbar-light"
      style={
        Object {
          "backgroundColor": "aliceblue",
        }
      }
    >
      <Link
        className="navbar-brand"
        to="/quizes"
      >
        Quiz Manager
      </Link>
      <button
        aria-controls="navbarText"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-target="#navbarText"
        data-toggle="collapse"
        style={
          Object {
            "backgroundColor": "whitesmoke",
          }
        }
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarText"
      >
        <ul
          className="navbar-nav mr-auto"
        >
          <li
            className="navbar-item"
          >
            <Link
              className="nav-link"
              to="/quizes"
            >
              Quizes
            </Link>
          </li>
          <li
            className="navbar-item"
          >
            <Link
              className="nav-link"
              onClick={[Function]}
              to="/"
            >
              Logout
            </Link>
          </li>
        </ul>
        <p>
          Welcome back, your access rights are:
           
          <b>
            Edit
          </b>
        </p>
      </div>
    </nav>
    <Switch>
      <Route
        exact={true}
        path="/login"
        render={[Function]}
      >
        <Redirect
          to="/quizes"
        />
      </Route>
      <Route
        path="/quizes"
        render={[Function]}
      />
      <Route
        path="/questions/:id"
        render={[Function]}
      />
      <Route
        path="/add-quizes"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/"
      >
        <Redirect
          to="/quizes"
        />
      </Route>
    </Switch>
  </div>
</BrowserRouter>
`;

exports[`App should match snapshot if user has restricted rights 1`] = `
<BrowserRouter>
  <div>
    <nav
      className="navbar navbar-expand-lg navbar-light"
      style={
        Object {
          "backgroundColor": "aliceblue",
        }
      }
    >
      <Link
        className="navbar-brand"
        to="/quizes"
      >
        Quiz Manager
      </Link>
      <button
        aria-controls="navbarText"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-target="#navbarText"
        data-toggle="collapse"
        style={
          Object {
            "backgroundColor": "whitesmoke",
          }
        }
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarText"
      >
        <ul
          className="navbar-nav mr-auto"
        >
          <li
            className="navbar-item"
          >
            <Link
              className="nav-link"
              to="/quizes"
            >
              Quizes
            </Link>
          </li>
          <li
            className="navbar-item"
          >
            <Link
              className="nav-link"
              onClick={[Function]}
              to="/"
            >
              Logout
            </Link>
          </li>
        </ul>
        <p>
          Welcome back, your access rights are:
           
          <b>
            Restricted
          </b>
        </p>
      </div>
    </nav>
    <Switch>
      <Route
        exact={true}
        path="/login"
        render={[Function]}
      >
        <Redirect
          to="/quizes"
        />
      </Route>
      <Route
        path="/quizes"
        render={[Function]}
      />
      <Route
        path="/questions/:id"
        render={[Function]}
      />
      <Route
        path="/add-quizes"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/"
      >
        <Redirect
          to="/quizes"
        />
      </Route>
    </Switch>
  </div>
</BrowserRouter>
`;

exports[`App should match snapshot if user has view rights 1`] = `
<BrowserRouter>
  <div>
    <nav
      className="navbar navbar-expand-lg navbar-light"
      style={
        Object {
          "backgroundColor": "aliceblue",
        }
      }
    >
      <Link
        className="navbar-brand"
        to="/quizes"
      >
        Quiz Manager
      </Link>
      <button
        aria-controls="navbarText"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-target="#navbarText"
        data-toggle="collapse"
        style={
          Object {
            "backgroundColor": "whitesmoke",
          }
        }
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarText"
      >
        <ul
          className="navbar-nav mr-auto"
        >
          <li
            className="navbar-item"
          >
            <Link
              className="nav-link"
              to="/quizes"
            >
              Quizes
            </Link>
          </li>
          <li
            className="navbar-item"
          >
            <Link
              className="nav-link"
              onClick={[Function]}
              to="/"
            >
              Logout
            </Link>
          </li>
        </ul>
        <p>
          Welcome back, your access rights are:
           
          <b>
            View
          </b>
        </p>
      </div>
    </nav>
    <Switch>
      <Route
        exact={true}
        path="/login"
        render={[Function]}
      >
        <Redirect
          to="/quizes"
        />
      </Route>
      <Route
        path="/quizes"
        render={[Function]}
      />
      <Route
        path="/questions/:id"
        render={[Function]}
      />
      <Route
        path="/add-quizes"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/"
      >
        <Redirect
          to="/quizes"
        />
      </Route>
    </Switch>
  </div>
</BrowserRouter>
`;

exports[`App should match snapshot if user is logged in 1`] = `
<BrowserRouter>
  <div>
    <nav
      className="navbar navbar-expand-lg navbar-light"
      style={
        Object {
          "backgroundColor": "aliceblue",
        }
      }
    >
      <Link
        className="navbar-brand"
        to="/quizes"
      >
        Quiz Manager
      </Link>
      <button
        aria-controls="navbarText"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-target="#navbarText"
        data-toggle="collapse"
        style={
          Object {
            "backgroundColor": "whitesmoke",
          }
        }
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarText"
      >
        <ul
          className="navbar-nav mr-auto"
        >
          <li
            className="navbar-item"
          >
            <Link
              className="nav-link"
              to="/quizes"
            >
              Quizes
            </Link>
          </li>
          <li
            className="navbar-item"
          >
            <Link
              className="nav-link"
              onClick={[Function]}
              to="/"
            >
              Logout
            </Link>
          </li>
        </ul>
        <p>
          Welcome back, your access rights are:
           
          <b />
        </p>
      </div>
    </nav>
    <Switch>
      <Route
        exact={true}
        path="/login"
        render={[Function]}
      >
        <Redirect
          to="/quizes"
        />
      </Route>
      <Route
        path="/quizes"
        render={[Function]}
      />
      <Route
        path="/questions/:id"
        render={[Function]}
      />
      <Route
        path="/add-quizes"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/"
      >
        <Redirect
          to="/quizes"
        />
      </Route>
    </Switch>
  </div>
</BrowserRouter>
`;

exports[`App should match snapshot if user is logged out 1`] = `
<BrowserRouter>
  <div>
    <nav
      className="navbar navbar-expand-lg navbar-light"
      style={
        Object {
          "backgroundColor": "aliceblue",
        }
      }
    >
      <Link
        className="navbar-brand"
        to="/quizes"
      >
        Quiz Manager
      </Link>
      <button
        aria-controls="navbarText"
        aria-expanded="false"
        aria-label="Toggle navigation"
        className="navbar-toggler"
        data-target="#navbarText"
        data-toggle="collapse"
        style={
          Object {
            "backgroundColor": "whitesmoke",
          }
        }
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        id="navbarText"
      >
        <ul
          className="navbar-nav mr-auto"
        >
          <li
            className="navbar-item"
          >
            <Link
              className="nav-link"
              to="/quizes"
            >
              Quizes
            </Link>
          </li>
          <li
            className="navbar-item"
          >
            <Link
              className="nav-link"
              to="/login"
            >
              Login
            </Link>
          </li>
        </ul>
      </div>
    </nav>
    <Switch>
      <Route
        exact={true}
        path="/login"
        render={[Function]}
      >
        <Login
          loginStateChange={[Function]}
        />
      </Route>
      <Route
        path="/quizes"
        render={[Function]}
      />
      <Route
        path="/questions/:id"
        render={[Function]}
      />
      <Route
        path="/add-quizes"
        render={[Function]}
      />
      <Route
        exact={true}
        path="/"
      >
        <Login
          loginStateChange={[Function]}
        />
      </Route>
    </Switch>
  </div>
</BrowserRouter>
`;
